compiler/modules/front.syntax/token.bal

```diff
@@ -487,9 +487,90 @@ public readonly class SourceFile {
         return unpackPosition(pos);
     }
 
-    public function lineContent(Position pos) returns string {
-        int lineNum = self.lineColumn(pos)[0];
-        return scanLineToString(self.scannedLine(lineNum));
+    // range is expected to be the start of a token
+    public function lineContent((Position|d:Range) range) returns [string, string, string] {
+        if range is Position {
```

Comment:
Why is the code for the Position case completely difference from the code for the Range case?

Reply:
If we have a range all we have to do is, turn the line in to a string and slice it because we know the start and end positions of all three prefix, content, and suffix. But with a position what we know is where the prefix ends. We have to determine where the content ends and suffix begins.

Reply:
But it's better to keep them more similar. For the case where there is no end Position, we can have a function

```
function tokenEndCodePointIndex(string[] lineContent, FragCode[] fragCodes, int tokenStartCodePointIndex) returns int;
```

Implement this using two functions similar to linePrefix and lineToken but have them return the codePointIndex of the end of the prefix and the end of the token respectively.

Order functions top down.